admin-ui/src/component/reservations/requested/hooks/index.ts

```diff
@@ -67,3 +76,127 @@ export const useReservationData = (
 
   return { ...rest, events };
 };
+
+const LIMIT = 100;
+
+type OptionsType = {
+  limit: number;
+};
+const defaultOptions = {
+  limit: LIMIT,
+};
+
+type CustomQueryParams = {
+  pk: number;
+  count: number;
+  offset: number;
+  state: ReservationsReservationStateChoices[];
+  begin?: Date;
+  end?: Date;
+};
+
+/// @param recurringPk fetch reservations related to this pk
+/// @param state optionally only fetch some reservation states
+/// @param limit allows to over fetch: 100 is the limit per query, larger amounts are done with multiple fetches
+///
+/// NOTE on cache
+/// refetching this query is a super bad idea
+/// use cache invalidation in the mutation instead
+/// The two solutions are either to queryInvalidate / refetch if you REALLY REALLY must do it (it's super slow)
+/// or use the update callback in the mutation to update the cache manually with cache.modify
+export const useRecurringReservations = (
+  recurringPk?: number,
+  options?: Partial<OptionsType>
+) => {
+  const { notifyError } = useNotification();
+  const { t } = useTranslation();
+
+  const states = [
+    ReservationsReservationStateChoices.Confirmed,
+    ReservationsReservationStateChoices.Denied,
+  ];
+  const { limit } = { ...defaultOptions, ...options };
+  const { data, loading, fetchMore } = useQuery<Query, CustomQueryParams>(
+    RECURRING_RESERVATION_QUERY,
+    {
+      skip: !recurringPk,
+      fetchPolicy: "cache-and-network",
+      nextFetchPolicy: "cache-first",
+      errorPolicy: "all",
+      variables: {
+        pk: recurringPk ?? 0,
+        offset: 0,
+        count: limit < LIMIT ? limit : LIMIT,
```

Comment:
Potentially confusing naming - maybe use `defaultOptions.limit` instead of the constant.